### PR TITLE
Cql spark evaluator monitoring spike

### PR DESCRIPTION
**Pull request by tabishop**
Below is the code needed to enable metrics output. Documentation will be added to confluence on how to import the dashboard and configure Prometheus. In addition, the following properties are required to be provided to spark-submit.

#Cohorting and Spark Metrics enablement properties
spark.metrics.conf.*.sink.prometheusServlet.class=org.apache.spark.metrics.sink.PrometheusServlet  <-- should be an asterisk between the 2 dots, but github comment isn't rendering it.
spark.metrics.conf.*.sink.prometheusServlet.path=/metrics/prometheus <-- should be an asterisk between the 2 dots, but github comment isn't rendering it.
spark.metrics.conf.master.sink.prometheusServlet.path=/metrics/master/prometheus
spark.metrics.conf.applications.sink.prometheusServlet.path=/metrics/applications/prometheus

spark.plugins=com.ibm.cohort.cql.spark.metrics.CustomMetricSparkPlugin

spark.kubernetes.driver.annotation.prometheus.io/scrape=true
spark.kubernetes.driver.annotation.prometheus.io/path=/metrics/prometheus/
spark.kubernetes.driver.annotation.prometheus.io/port=4040

spark.kubernetes.driver.service.annotation.prometheus.io/scrape=true
spark.kubernetes.driver.service.annotation.prometheus.io/path=/metrics/executors/prometheus/
spark.kubernetes.driver.service.annotation.prometheus.io/port=4040

spark.ui.prometheus.enabled=true
spark.executor.processTreeMetrics.enabled=true
spark.sql.streaming.metricsEnabled=true
spark.metrics.appStatusSource.enabled=true

### CODE REVIEW

File: cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/StringGauge.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.ibm.cohort.cql.spark;
```
Comment (csandersdev) [inline cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/StringGauge.java:1]: Needs copyright

**Review by csandersdev — COMMENTED** [review]
Could you elaborate on why you are materializing specific typed versions of the Gauge generic (StringGauge , IntGauge , LongGauge )? Not that it hurts anything. It just seems like unnecessary class files to maintain.

Outside the one copyright issue and DCO failures, otherwise LGTM.

**Comment by tabishop** [comment]
The DropWizard package just includes an untyped interface for Gauge and you have to implement your own. I suppose I could have done something generic with just Object, but it was only 1 extra class for strong typing.

File: metrics/grafana/cohortGrafanaDashboard.json [hunk 2]
```diff
@@ -0,0 +1,987 @@
+{
```
Comment (msargentibm) [inline metrics/grafana/cohortGrafanaDashboard.json:1]: Will the documentation contain a note on how this file was created?

**Pull request closed without merging**